src/main/webapp/app/admin/metrics/metrics.component.ts

```diff
@@ -26,32 +26,29 @@ export class JhiMetricsMonitoringComponent implements OnInit {
     /**
      * Refreshes the metrics by retrieving all metrics and thread dumps
      */
-    refresh() {
+    refresh(): void {
         this.updatingMetrics = true;
-        this.metricsService.getMetrics().subscribe((metrics) => {
+        combineLatest([this.metricsService.getMetrics(), this.metricsService.threadDump()]).subscribe(([metrics, threadDump]) => {
             this.metrics = metrics;
-            this.metricsService.threadDump().subscribe((data) => {
-                this.threadData = data.threads;
-                this.updatingMetrics = false;
-            });
+            this.threads = threadDump.threads;
+            this.updatingMetrics = false;
+            this.changeDetector.markForCheck();
         });
     }
 
     /**
      * Checks if the metric with the key {@param key} exists
-     * @param metrics json with metrics objects
      * @param key string identifier of a metric
      */
-    isObjectExisting(metrics: any, key: string) {
-        return metrics && metrics[key];
+    metricsKeyExists(key: keyof Metrics): boolean {
+        return Boolean(this.metrics?.[key]);
     }
 
     /**
      * Checks if the metric with the key {@param key} exists and is not empty
-     * @param metrics json with metrics objects
      * @param key key string identifier of a metric
      */
-    isObjectExistingAndNotEmpty(metrics: any, key: string) {
-        return this.isObjectExisting(metrics, key) && JSON.stringify(metrics[key]) !== '{}';
+    metricsKeyExistsAndObjectNotEmpty(key: keyof Metrics): boolean {
+        return Boolean(this.metrics?.[key] && JSON.stringify(this.metrics[key]) !== '{}');
```

Comment:
Either rename to `metricsKeyExistsAndIsNonEmptyObject()` or change the second statement to compare to the size of the object being greater than 0 (e.g. by taking the values array)

Reply:
Not necessary since generated from jhipster